xunit-autolabeler-v2/ast_parser/python/test_parser.py

```diff
@@ -0,0 +1,225 @@
+# Copyright 2020 Google LLC. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import ast
+import os
+import sys
+from typing import Any, Dict, List, Tuple
+
+
+from ast_parser.core import constants
+from . import drift_test
+
+
+def _get_test_nodes(parsed_nodes: List[Any]) -> List[Any]:
+    """Helper function to extract test methods wrapped in classes
+
+    Args:
+        parsed_nodes (List[ast.AST]): a list of nodes parsed
+                                      from a Python test file
+
+    Returns:
+        List[ast.AST]: a list of possible test nodes
+    """
+
+    class_nodes = [node for node in parsed_nodes
+                   if 'ClassDef' in str(type(node))]
+    test_nodes = [node for node in parsed_nodes
+                  if node not in class_nodes]
+
+    for node in class_nodes:
+        test_nodes += node.body
+
+    test_nodes = [node for node in test_nodes
+                  if hasattr(node, 'name') and
+                  node.name.startswith('test_')]
+
+    return test_nodes
+
+
+def get_test_methods(test_path: str) -> List[Any]:
+    """Gets the top-level methods within a test file
+
+    Args:
+        source_path: path to the file to process
+
+    Returns:
+        List[ast.AST]: a list of the top-level
+                       methods within the provided file
+    """
+    try:
+        with open(test_path, 'r') as file:
+            content = "".join(file.readlines())
```

Comment:
Nit: do you want to consistently use single quotes?

Reply:
Fixed. (In a future PR, I'll add a single-quote linter to `flake8`.)

Reply:
I still see the double quotes.

```python
content = "".join(file.readlines())
```

Reply:
Fixed.